src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue

```diff
@@ -0,0 +1,239 @@
+                :rules="rules.fromDate"
+                required
+              />
+            </oxd-grid-item>
+            <oxd-grid-item>
+              <date-input
+                v-model="review.dueDate"
+                :label="$t('performance.due_date')"
+                :rules="rules.fromDate"
+                required
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+        <oxd-divider />
+        <oxd-form-actions>
+          <required-text />
+          <div class="orangehrm-button-row">
+            <oxd-button
+              display-type="ghost"
+              :label="$t('general.cancel')"
+              type="button"
+              @click="onCancel"
+            />
+            <oxd-button
+              display-type="ghost"
+              :label="$t('general.save')"
+              type="submit"
+              @click="onSave(false)"
+            />
+            <oxd-button
+              class="orangehrm-left-space"
+              display-type="secondary"
+              :label="$t('performance.activate')"
+              type="button"
+              @click="onSave(true)"
+            />
+          </div>
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {navigate} from '@/core/util/helper/navigation';
+import EmployeeAutocomplete from '@/core/components/inputs/EmployeeAutocomplete';
+import SupervisorAutoComplete from '@/orangehrmPerformancePlugin/components/SupervisorAutoComplete';
+import {APIService} from '@/core/util/services/api.service';
+import {
+  required,
+  startDateShouldBeBeforeEndDate,
+  validDateFormat,
+} from '@/core/util/validation/rules';
+import usei18n from '@/core/util/composable/usei18n';
+import {ref} from 'vue';
+
+const reviewModel = {
+  employee: null,
+  supervisorReviewer: null,
+  startDate: null,
+  endDate: null,
+  dueDate: null,
+};
+
+export default {
+  components: {
+    'employee-autocomplete': EmployeeAutocomplete,
+    'supervisor-autocomplete': SupervisorAutoComplete,
+  },
+  props: {
+    reviewId: {
+      type: Number,
+      required: true,
+    },
+  },
+  setup() {
+    const review = ref({...reviewModel});
+    const {$t} = usei18n();
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      '/api/v2/performance/reviews',
+    );
+    http.setIgnorePath('/api/v2/performance/reviews/[0-9]+');
+    const rules = {
+      employee: [required],
+      supervisorReviewer: [required],
+      startDate: [
+        required,
+        validDateFormat(),
+        startDateShouldBeBeforeEndDate(
+          () => review.value.endDate,
+          $t('general.from_date_should_be_before_to_date'),
+          {allowSameDate: false},
+        ),
+      ],
+      endDate: [required],
+      dueDate: [required],
+    };
+    return {
+      http,
+      rules,
+    };
+  },
+  data() {
+    return {
+      isLoading: false,
+      review: {...reviewModel},
```

Comment:
you declared review inside `setup()` why re-declare here? add it to return block of `setup()`

Reply:
also move `isLoading` as a ref also. then can remove `data()`

Reply:
I'll move the rules to data to keep it same as AddReview